02-peer-protocol.md

```diff
@@ -2439,6 +2948,44 @@ such detection is left as an option.
 Nodes inside a blinded route must use `invalid_onion_blinding` to avoid
 leaking information to senders trying to probe the blinded route.
 
+### Batching channel messages
+
+Multiple channel messages can be grouped together as a single logical message
+using the `start_batch` message.
+
+1. type: 127 (`start_batch`)
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`u16`:`batch_size`]
+
+#### Requirements
+
+A sending node:
+  - MUST set `batch_size` to a value strictly greater than 1.
+  - MUST set `batch_size` to a value strictly lower than 20.
+  - After sending `start_batch`:
+    - MUST send `batch_size` messages for the same `channel_id`, without any
+      other unrelated messages in-between.
+
+A receiving node:
+  - If `batch_size` is not strictly greater than 1:
+    - MUST ignore the `start_batch` message.
+    - SHOULD send a `warning`.
+  - If `batch_size` is greater than or equal to 20:
+    - MUST send a `warning` and close the connection, or send an `error` and
+      fail the channel.
+  - MUST group the next `batch_size` messages and process them together.
+  - If one of those messages is not for the specified `channel_id`:
+    - MUST send a `warning` and close the connection, or send an `error` and
+      fail the channel.
+
+#### Rationale
+
+We limit the `batch_size` to 20 elements to protect against excessive queuing
+that could be abused to DoS receiving nodes. The `start_batch` message is only
+used for splice RBF attempts so far: 20 RBF attempts should be enough to get
```

Comment:
Wouldn't this by 19 RBF attempts since there is one `commitment_signed` for the confirmed `funding_tx`?